docs/examples/mht_example.py

```diff
@@ -0,0 +1,254 @@
+# multiple objects and handling uncertainties and ambiguities of tracks (e.g. presence of
+# clutter).
+#
+# MHT, by definition, has several algorithms that fall under this definition, which
+# include Global Nearest Neighbour
+# (GNN, tutorial `here <https://stonesoup.readthedocs.io/en/v1.1/auto_tutorials/06_DataAssociation-MultiTargetTutorial.html#sphx-glr-auto-tutorials-06-dataassociation-multitargettutorial-py>`__),
+# Joint Probabilistic Data association
+# (JPDA, tutorial `here <https://stonesoup.readthedocs.io/en/v1.1/auto_tutorials/08_JPDATutorial.html#sphx-glr-auto-tutorials-08-jpdatutorial-py>`__),
+# Multi-frame assignment
+# (MFA [#]_, see other example `here <https://stonesoup.readthedocs.io/en/v1.1/auto_examples/MFA_example.html#sphx-glr-auto-examples-mfa-example-py>`__),
+# Multi Bernoulli filter and Probabilistic multi hypotheses tracking (PMHT).
+# Some of these algorithms are already implemented the Stone Soup.
+# In this example we employ the multi-frame assignment data associator and
+# hypothesiser using their Stone Soup implementation.
+#
+# This example follows this structure:
+#   1. Create ground truth and detections;
+#   2. Instantiate the tracking components and tracker;
+#   3. Run the tracker and visualise the results.
+#
+
+# %%
+# General imports
+# ^^^^^^^^^^^^^^^
+import numpy as np
+from datetime import datetime, timedelta
+from itertools import tee
+
+# %%
+# Stone Soup imports
+# ^^^^^^^^^^^^^^^^^^
+from stonesoup.types.array import StateVector, CovarianceMatrix
+from stonesoup.types.state import GaussianState
+from stonesoup.models.transition.linear import CombinedLinearGaussianTransitionModel, \
+    ConstantVelocity
+from stonesoup.models.measurement.nonlinear import CartesianToBearingRange
+from stonesoup.simulator.simple import MultiTargetGroundTruthSimulator, SimpleDetectionSimulator
+
+# Simulation parameters
+np.random.seed(1908)  # fix a random seed
+start_time = datetime.now().replace(microsecond=0)
+simulation_steps = 50
+timestep_size = timedelta(seconds=2)
+prob_detection = 0.99
+initial_state_mean = StateVector([[10], [0], [10], [0]])
+initial_covariance = CovarianceMatrix(np.diag([30, 1, 40, 1]))
+
+# clutter will be generated uniformly in this area around the targets
+clutter_area = np.array([[-1, 1], [-1, 1]])*150
+clutter_rate = 9
+surveillance_area = ((clutter_area[0][1] - clutter_area[0][0])*
+                     (clutter_area[1][1] - clutter_area[1][0]))
+clutter_spatial_density = clutter_rate/surveillance_area
+
+# %%
+# 1. Create ground truth and detections;
+# --------------------------------------
+# We have prepared all the general parameters for the simulation,
+# including the clutter spatial density. In this example we set
+# the birth rate and the death probability as zero, using only the knowledge of the
+# prior states to generate the tracks so the number of targets is fixed (3 in this case).
```

Comment:
Am I right in thinking that MFA (or this implementation) only works with fixed number of tracks?